modules/brew/README.md

```diff
@@ -0,0 +1,36 @@
+# brew
```

Comment:
# Nofile limits

I think that we should document the necessary tweak to install/run some brew packages. Else, some packages complain about file limits & fail to run.

It is necessary to increase nofile limits, like it's done here (both limits.d & systemd config should be modified):


- SSH/tty nofile limit
`/usr/etc/security/limits.d/30-brew-limits.conf`

```
#This file sets the resource limits for the users logged in via PAM,
#more specifically, users logged in on via SSH or tty (console).
#Limits related to terminals in Wayland/Xorg sessions depend on a
#change to /etc/systemd/user.conf.
#This does not affect resource limits of the system services.
#This file overrides defaults set in /etc/security/limits.conf

* soft nofile 4096
* hard nofile 524288
```

- SystemD nofile limit affecting whole session
`/usr/lib/systemd/system/system.conf.d/30-brew-limits.conf`
`/usr/lib/systemd/user/user.conf.d/30-brew-limits.conf`

```
[Manager]
DefaultLimitNOFILE=4096:524288
```

# Necessary dependencies

Document that GCC needs to be installed in the image for brew to work.
I read that brew can provide gcc too, but I would let OS package manager to handle this one, since GCC is available in basically all distros.

Reply:
Should we add those overrides as lines of `cat` in brew.sh, the same way we did for the systemd units? 


> # Necessary dependencies
> Document that GCC needs to be installed in the image for brew to work. I read that brew can provide gcc too, but I would let OS package manager to handle this one, since GCC is available in basically all distros.

Recommended way: 
```
rpm:
  install:
    - gcc
```

Reply:
I'm not sure if we should automatically add those for all users, since increasing nofile limits can have some potential security implications if application chooses to abuse it for evil purposes. I'm not the security expert, so can't explain it in detail, but that's how I see it.

It can also potentially override the same modification that someone did differently in their image for some other purposes.

I think that just recommendation in docs is better.

About GCC, I think that we can check if `gcc` command exists. If it doesn't, than inform the user that `gcc` is not installed, how it must be installed for brew to function & fail the build.

Reply:
I think these two points can be resolved by improving brew.sh.
We could a flag that manages nofile-limits? It would default to False, and clearly mention the potential security implications if True.

Reply:
I'll let others also discuss this review discussion & than try to find the sweet-spot solution.